simulator/lang/README.md

```diff
@@ -0,0 +1,569 @@
+
+Scenarios need an `init` function in addition to the constructor, in order to perform `async` loading of the test environment data.
+
+```javascript
+await is.init();
+```
+
+Prior to running this JavaScript code, we started the SimServer with the command `reach compile --simulate`. We can ping the server to confirm that it is up and running.
+
+```javascript
+const pi = await is.pingServer();
+```
+
+The `Scenario` has information about the participants in the current simulation.
+
+```javascript
+const alice = is.participants.Alice;
+const bob = is.participants.Bob;
+```
+
+As well as the consensus.
+
+```javascript
+const consensus = is.consensus;
+```
+
+Participants need an `init` function in addition to the constructor, in order to perform `async` loading of the test participant data.
+
+```javascript
+const [, a] = await alice.init();
+const [s, b] = await bob.init();
+```
+
+After designing and building a multiplayer game such as _Rock, Paper, Scissors_, a straightforward approach to testing the game comprehensively might involve running the game with all possible inputs from each player and asserting the outcomes. SimLang makes this process easy and convenient by allowing us to _branch state_ at this point in the simulation and run the game for all possible inputs using the `play` function as seen below. `play` returns the actual outcome of the game, which we then compare against our expected outcome.
+
+```javascript
+for (let aHand = 0; aHand < 3; aHand++) {
+  for (let bHand = 0; bHand < 3; bHand++) {
+    const r = await play(s.copy(),aHand,bHand,a,b,consensus);
+    r.assertVar(winner(aHand,bHand));
+  }
+}
+```
+
+The `play` function is defined as follows. It accepts:
+
+1. a reference to a Scenario `sc`
+2. Alice's input/hand `aHand`
+3. Bob's input/hand `bHand`
+4. a reference to the participant `alice`
+5. a reference to the participant `bob`
+6. a reference to the  `consensus`
+
+In this example, we are simulating the most basic form of the game _Rock, Paper, Scissors_. Alice plays her hand, and it is published by the consensus. Then Bob plays his hand, and it is published by the consensus. Both players learn the outcome of the game, and subsequently exit the decentralized application.
+
+```javascript
+const play = async (sc,aHand,bHand,alice,bob,consensus) => {
+  let s = await sc.who(alice).interact('getHand', aHand);
+  s = await s.who(consensus).publish(alice);
+  s = await s.who(bob).interact('getHand', bHand);
+  await consensus.publish(bob);
+  await alice.exit();
+  await bob.exit();
+  const r = await alice.getStatus();
+  // game over
+  assert.equal(r,"Done");
+  return alice.getVar('outcome');
+}
+```
+
+## rps-6-timeouts
+In this section of the tutorial, naturally we are inclined to demonstrate exploring timeouts with SimLang. For convenience, rather than having to think about the clock when attempting to test timeout situations, the SimLang library provides a `forceTimout` method with the `Scenario` object.
```

Comment:
```suggestion
In this section of the tutorial, naturally we are inclined to demonstrate exploring timeouts with SimLang. For convenience, rather than having to think about the clock when attempting to test timeout situations, the SimLang library provides a `forceTimeout` method with the `Scenario` object.
```